src/firefly/js/visualize/draw/ShapeDataObj.js

```diff
@@ -706,39 +731,46 @@ export function drawText(drawObj, ctx, plot, inPt, drawParams) {
         }
 
         angle = pvAngle + lineAngle;
+    } else {
+        const pv = getPlotViewById(visRoot(), plot.plotId);
+        angle = pv.flipY? 180 - pv.rotation:pv.rotation;
```

Comment:
I don't think the flip logic is right. I flipped an image and used the distance tool  and the text was not correct.